clusterloader2/pkg/measurement/common/etcd_metrics.go

```diff
@@ -136,7 +154,11 @@ func (e *etcdMetricsMeasurement) stopAndSummarize(host, provider string) error {
 	if err != nil {
 		return err
 	}
-	for _, sample := range samples {
+
+	collectEtcdMetrics := func(sample *model.Sample) {
+		e.Lock()
+		defer e.Unlock()
+
```

Comment:
Alternatively you could also refactor this to first in switch find address to the HistogramVec you should be modifying and then have a short Lock-Unlock block where you would call `measurementutil.ConvertSampleToBucket(sample, histogramVec)`

Reply:
Done.